lib/datadog/ci/ext/environment.rb

```diff
@@ -211,11 +214,23 @@ def extract_buildkite(env)
             Core::Git::Ext::TAG_COMMIT_AUTHOR_NAME => env['BUILDKITE_BUILD_AUTHOR'],
             Core::Git::Ext::TAG_COMMIT_AUTHOR_EMAIL => env['BUILDKITE_BUILD_AUTHOR_EMAIL'],
             Core::Git::Ext::TAG_COMMIT_MESSAGE => env['BUILDKITE_MESSAGE'],
+            TAG_NODE_NAME => env['BUILDKITE_AGENT_ID'],
             TAG_CI_ENV_VARS => {
               'BUILDKITE_BUILD_ID' => env['BUILDKITE_BUILD_ID'],
               'BUILDKITE_JOB_ID' => env['BUILDKITE_JOB_ID']
             }.to_json
           }
+
+          extra_tags = env
+            .select { |key, value| key.start_with?('BUILDKITE_AGENT_META_DATA_') }
+            .map { |key, value| "#{key.to_s.sub('BUILDKITE_AGENT_META_DATA_', '').downcase}:#{value}"}
+            .sort_by { |key| key.length }
+
+          if extra_tags.length > 0
+            tags[TAG_NODE_LABELS] = JSON.generate(extra_tags)
```

Comment:
`JSON.generate(extra_tags)` and `extra_tags.to_json` are analogous. Since it looks like we are using `to_json` in other parts of this file, should we try being consistent and use `to_json` here as well 😄 ?

Reply:
I tried that first, but it seems `.to_json` does not work for lists. `.to_s` seems to work but it adds some extra spaces I don't want: maybe there's a workaround?

Reply:
Interesting

I just did a simple example in `irb`

```
require 'json'
=> true
irb(main):002:0> test = ["mytag:my-value", "myothertag:my-other-value"]
=> ["mytag:my-value", "myothertag:my-other-value"]
irb(main):003:0> test.to_json
=> "[\"mytag:my-value\",\"myothertag:my-other-value\"]"
```

That is what you want, right?

Reply:
ok I messed it up somehow 😆 . I'll change it, thanks